flink-ml-lib/src/test/java/org/apache/flink/ml/feature/BucketizerTest.java

```diff
@@ -0,0 +1,190 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/** Tests the {@link Bucketizer}. */
+public class BucketizerTest extends AbstractTestBase {
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+    private StreamTableEnvironment tEnv;
+    private Table inputTable;
+
+    private static final List<Row> inputData =
+            Arrays.asList(
+                    Row.of(1, -0.5, 0.0, 1.0),
+                    Row.of(2, Double.NEGATIVE_INFINITY, 1.0, Double.POSITIVE_INFINITY),
+                    Row.of(3, Double.NaN, -0.5, -0.5));
+
+    private static final Double[][] splitsArray =
+            new Double[][] {
+                new Double[] {-0.5, 0.0, 0.5},
+                new Double[] {-1.0, 0.0, 2.0},
+                new Double[] {Double.NEGATIVE_INFINITY, 10.0, Double.POSITIVE_INFINITY}
+            };
+
+    private final List<Row> expectedKeepResult =
+            Arrays.asList(Row.of(1, 0, 1, 0), Row.of(2, 2, 1, 1), Row.of(3, 2, 0, 0));
+
+    private final List<Row> expectedSkipResult = Collections.singletonList(Row.of(1, 0, 1, 0));
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+        inputTable = tEnv.fromDataStream(env.fromCollection(inputData)).as("id", "f1", "f2", "f3");
+    }
+
+    @SuppressWarnings("all")
+    private void verifyOutputResult(Table output, String[] outputCols, List<Row> expectedResult)
+            throws Exception {
+        List<Row> collectedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        List<Row> result = new ArrayList<>(collectedResult.size());
+
+        for (Row r : collectedResult) {
+            Row outRow = new Row(outputCols.length + 1);
+            outRow.setField(0, r.getField("id"));
+            for (int i = 0; i < outputCols.length; i++) {
+                outRow.setField(i + 1, r.getField(outputCols[i]));
+            }
+            result.add(outRow);
+        }
+
+        compareResultCollections(
+                expectedResult, result, Comparator.comparingInt(r -> ((Integer) r.getField(0))));
+    }
+
+    @Test
+    public void testParam() {
+        Bucketizer bucketizer = new Bucketizer();
+        assertEquals(HasHandleInvalid.ERROR_INVALID, bucketizer.getHandleInvalid());
+
+        bucketizer
+                .setInputCols("f1", "f2", "f3")
+                .setOutputCols("o1", "o2", "o3")
+                .setHandleInvalid(HasHandleInvalid.SKIP_INVALID)
+                .setSplitsArray(splitsArray);
```

Comment:
@lindong28 How do you like the idea to add APIs to `Bucketizer` so that users can define metadata in the following format?
```java
bucketizer
    .addCol("f1", "o1", splitsArray[0])
    .addCol("f2", "o2", splitsArray[1]);
```
It is different from existing practice from Spark or Alink, but the readability would better. It would also be OK for me if it is better to leave this discussion to future and to focus on functionality correctness for now.

Reply:
It seems better keep the current approach so that the user-experience with `setInputCols(..)` names is more consistent with `setInputCol(...)`.

IMO, it is not clear that the readability with the alternative approach is better than the current approach.